src/vs/workbench/api/common/extHostTreeViews.ts

```diff
@@ -406,6 +415,16 @@ class ExtHostTreeView<T> extends Disposable {
 		this.proxy.$setTitle(this.viewId, this._title, description);
 	}
 
+	private _badge: vscode.ViewBadge | undefined;
+	get badge(): vscode.ViewBadge | undefined {
+		return this._badge;
+	}
+
+	set badge(badge: vscode.ViewBadge | undefined) {
+		this._badge = badge;
+		this.proxy.$setBadge(this.viewId, badge);
```

Comment:
I am passing the `vscode.ViewBadge` directly through to the proxy here (as the proxy's `IViewBadge` interface matches `vscode.ViewBadge`).  I would imagine that the proxy JSON-serializes this object and so this would work fine, but if it's better to make a new object here, or otherwise treat this data differently, let me know.

Reply:
The better way is to copy all expected properties from `vscode.ViewBadge` into the target shape/object. In theory extensions can give you an object which more than `ViewBadge` containing lots of data or cycles

Reply:
Done.